draft-ietf-rtcweb-mdns-ice-candidates.md

```diff
@@ -345,8 +366,45 @@ IP addresses that this specification is designed to hide. The use of
 registered mDNS hostnames SHOULD be scoped by origin, and SHOULD have the
 lifetime of the page.
 
+Determination of Address Privacy
+--------------------------------
+
+Naturally, an address that is already exposed to the Internet does not need to
+be protected by mDNS, as it can be trivially observed by the web server or
+remote endpoint. However, determining this ahead of time is not straightforward;
+while the fact that an IPv4 address is private can sometimes be inferred by its
+value, e.g., whether it is an {{RFC1918}} address, the reverse is not
+necessarily true. IPv6 addresses present their own complications, e.g.,
+private IPv6 addresses as a result of NAT64 {{RFC6146}}.
+
+Instead, the determination of whether an address is public can be reliably made
+as part of the ICE gathering process, namely, if the query to the
+STUN {{RFC5389}} server returns the same value as the local address. This can
+be done for both IPv4 and IPv6 local addresses, provided that the application
+has configured both IPv4 and IPv6 STUN servers. If this situation occurs, i.e.,
+STUN returns the same IP address value for an address that has already
+been communicated as a mDNS candidate during the current ICE gathering phase,
+the ICE agent MUST NOT eliminate the candidate as redundant and MUST send
+the IP address as a server-reflexive candidate. This allows the ICE agent
+to send mDNS candidates immediately, even if the associated addresses may not
```

Comment:
I had some confusion when reading this line. "this situation" in line 384 refers to the case when the mDNS candidate has already been communicated with this public address, and does "allow the ICE agent to send mDNS candidates immediately" refers to the subsequent caching in the next paragraph?

Reply:
Ah, I see what you mean. I made a suggestion below, maybe too verbose. Feel free to discard it or change to something similar.
```suggestion
to send mDNS candidates immediately without waiting for STUN responses, even if the associated addresses may not
```